x-pack/plugins/enterprise_search/public/applications/shared/query_params/query_params.test.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+jest.mock('queryString', () => ({
+  ...(jest.requireActual('queryString') as object),
+  parse: jest.fn(),
+}));
+
+import queryString from 'query-string';
+
+import { parseQueryParams } from './';
+
+describe('parseQueryParams', () => {
+  it('should call queryString parse method', () => {
+    const parseMock = jest.fn();
+    jest.spyOn(queryString, 'parse').mockImplementation(parseMock);
```

Comment:
you shouldn't need both `jest.spyOn` and `jest.mock`+import up above.

I'm a little confused also, why mock queryString? Does it return an expected `{ foo: 'bar' }` obj when calling `parseQueryParams('?foo=bar')`?

Reply:
When I don't mock it I get this error:

```
expect(received).toHaveBeenCalledWith(...expected)

Matcher error: received value must be a mock or spy function
```

Reply:
Right, but I'm saying don't bother spying on it, just evaluate the outcome since it's fairly straightforward.

Reply:
Got it! Thanks for the help. TIL actually testing the output of a library in a test 😂

Reply:
Haha, it doesn't always work out with the more complicated libraries but in the case it's definitely worth doing. Also you totally beat my other comment, jinx :D